editors/code/src/inlay_hints.ts

```diff
@@ -163,12 +164,20 @@ class HintsUpdater {
 
         this.pending.set(documentUri, tokenSource);
         try {
-            return await sendRequestWithRetry<InlayHint[] | null>(
+            return await sendRequestWithRetry<InlayHint[]>(
                 client,
                 'rust-analyzer/inlayHints',
                 request,
                 tokenSource.token,
             );
+        } catch (err) {
+            if (!isCancellationError(err)) {
+                // FIXME: log the error
+                throw err;
+            }
+
+            assert(tokenSource.token.isCancellationRequested);
+            return null;
```

Comment:
I suggest not discriminating cancellation errors specifically:

https://github.com/microsoft/vscode-languageserver-node/blob/f425af9de46a0187adb78ec8a46b9b2ce80c5412/client/src/colorProvider.ts#L66-L69

Reply:
@matklad, but cancellation errors are not actually errors, they don't even deserve a line our logs... I agree on catching **all** errors, since otherwise they will propagate and result into `unhandled promise rejections`. but cancellations are too pervasive and will flood error logs otherwise